weaviate/connection/rest.go

```diff
@@ -53,6 +54,39 @@ func NewConnection(scheme string, host string, httpClient *http.Client, headers
 	return connection
 }
 
+// WaitForWeaviate waits until weaviate is started up and ready
+func (con *Connection) WaitForWeaviate(startup_period int) error {
+	if startup_period < 0 {
+		return errors.New("'startup_period' needs to be an integer larger than zero")
+	}
+	ticker := time.NewTicker(time.Second)
+	start_time := time.Now()
+	for {
+		t := <-ticker.C
+		ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second)
+		response, err := con.RunREST(ctx, "/.well-known/ready", http.MethodGet, nil)
+		var isReady bool
+		switch {
+		case err != nil:
+			isReady = false
+		case response.StatusCode == 200:
+			isReady = true
+		default:
+			isReady = false
+
+		}
+
+		cancelFunc()
+		if isReady {
+			return nil
+		}
+		if t.After(start_time.Add(time.Duration(startup_period) * time.Second)) {
```

Comment:
Looking at this, I think it would be nicer to have this method's parameter type as a `time.Duration` rather than an int